esp-hal/src/gpio/mod.rs

```diff
@@ -1044,57 +1033,36 @@ fn on_pin_irq(pin_nr: u8) {
     asynch::PIN_WAKERS[pin_nr as usize].wake(); // wake task
 }
 
-#[doc(hidden)]
-pub trait GpioProperties {
-    type Bank: BankGpioRegisterAccess;
-    type InterruptStatus: InterruptStatusRegisterAccess;
-
-    type IsOutput: BooleanType;
-    type IsAnalog: BooleanType;
-    type IsTouch: BooleanType;
-
-    fn degrade_pin(&self, _: private::Internal) -> AnyPin;
-    fn output_signals() -> [Option<OutputSignal>; 6];
-    fn input_signals() -> [Option<InputSignal>; 6];
-}
-
 #[doc(hidden)]
 #[macro_export]
 macro_rules! if_output_pin {
-    (InputOnlyAnalog, { $($then:tt)* } else { $($else:tt)* } ) => { $($else)* };
     (InputOutputAnalog, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
     (InputOutputAnalogTouch, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
     (InputOutput, { $($then:tt)* } else { $($else:tt)* } ) => { $($then)* };
+    ($other:ident, { $($then:tt)* } else { $($else:tt)* } ) => { $($else)* };
 }
 pub(crate) use if_output_pin;
 
 #[doc(hidden)]
 #[macro_export]
-macro_rules! pin_types {
-    (InputOnly) => {
-        type IsOutput = $crate::gpio::False;
-        type IsAnalog = $crate::gpio::False;
-        type IsTouch = $crate::gpio::False;
+macro_rules! io_types {
```

Comment:
I guess the rename doesn't align with what I want to do long term, but whatever. Long term I want to declare all pin type traits (Analog, Touch, LP...) here, in such a way that we can move the per-chip implementation to the soc module where they belong.